### DISCUSSION TITLE
Silent Install Option for Windows Installer

### DISCUSSION THREAD

**Discussion by BDaddyG** [keepassxreboot/keepassxc#6107] [Q&A]
Hello everyone,

We're using Keepass-XC in a small Citrix/RDS environment (6 Windows 2012 R2 servers, 450 users).  We are not using any sort of software package management so we manually installed software and software updates.

Is there a way of having a silent install that automatically accepts the user agreement, deselects "Start Keepass-XC at login", and deselects "Start Keepass-XC now" at the end of the install steps?

Having this option would save us some mouse clicks.

Thanks!!

**Comment by droidmonkey** [comment]
This should get you started http://www.itninja.com/blog/view/the-complete-guide-to-msi-switches-for-silent-software-installation

Here are the name/value pairs https://github.com/keepassxreboot/keepassxc/blob/af55d1b1b364f222e13c34ec46efe22b83229860/share/windows/wix-template.xml#L88

**Reply by BDaddyG** [reply]
Fantastic!

Thanks for the links for the msi switches and name/value pairs!  I've used the msi switches before but the name/value pairs is new for me.

Have a great day!

**Reply by BDaddyG** [reply]
I've worked to get the name/value pairs to be accepted correctly but no luck.

I've tired the following:

msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM=0 LAUNCHAPPONEXIT=0 
msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM=No LAUNCHAPPONEXIT=No
msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM="0" LAUNCHAPPONEXIT="0"
msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM="No" LAUNCHAPPONEXIT="No" 
msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM=0
msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM="0"
msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM="No"

but an autorun entry is created in the registry at "HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Run" each time.

In the log file I can see the values I'm using on the command line so they are being passed:
For example I tired msiexec /l* c:\temp\keepassxc-264-log.txt /i KeePassXC-2.6.4-Win64.msi AUTOSTARTPROGRAM="BlankEntry"

and I found at line 70:
Property(S): AUTOSTARTPROGRAM = BlankEntry

Key to the switches I'm using:
/l* c:\temp\keepassxc-264-log.txt = (Lowercase L) log everything to the file c:\temp\keepassxc-264-log.txt
/i = input MSI file KeePassXC-2.6.4-Win64.msi
/q = quite, no dialog UI, this accepts/bypasses the user agreement and prevents KeePass-XC from starting at the end of the install

Any thoughts as to what I might be doing wrong?

**Reply by droidmonkey** [reply]
I just tried myself, and of course WIX being the most counterintuitive thing in the world, a setting is missing from the property. The properties should have been defined as:

```
<Property Id="AUTOSTARTPROGRAM" Value="1" Secure="yes" />
<Property Id="LAUNCHAPPONEXIT" Value="1" Secure="yes" />
<Property Id="INSTALLDESKTOPSHORTCUT" Secure="yes" />
```

Which of course, Secure makes so much sense as to allowing you to override the property.....

If you want I can provide a custom build of 2.6.4 msi with this fixed.

**Reply by BDaddyG** [reply]
This is great!  Thanks for the quick turn around!

Reviewing the above Property Id for INSTALLDESKTOPSHORTCUT, should it have Value="1" also?  If not, what is the command line option to disable the desktop shortcut creation?

I'd be happy to test a new build if that helps.  Otherwise I can wait and try this again with the next release.

**Reply by droidmonkey** [reply]
We have INSTALLDESKTOPSHORTCUT disabled by default. In WIX that means _no value_ (ie, undefined). If you set it to 0 that is a value and will check the checkbox.

**Reply by BDaddyG** [reply]
Thanks for the further detail. We don't want the desktop shortcut so I'll not be setting it.

**Reply by phoerious** [reply]
Unwanted desktop shortcuts. I can think of hardly anything that is more annoying. 🙄

**Reply by ReanimationXP** [reply]
I am attempting to silently deploy to users who are not able to update the application themselves as they are not admins. Is there a flag, or could one be added, to prevent the user from receiving the option to enable auto updates on first launch?  Presumably this would constantly nag them about new updates when they aren't able to install anything.

If there is not an install switch for this (yet), is there a registry key I could add post-install to prevent the question on launch?

**Reply by droidmonkey** [reply]
No but it is requested. The user can easily disable update notices in the settings.

**Reply by ReanimationXP** [reply]
Ok, where is that setting stored, such that I can manually disable the update checks?

**Reply by droidmonkey** [reply]
In the local config file, `%APPDATALOCAL%/keepassxc/keepassxc.ini`
